tests/integration/test_array_creation.py

```diff
@@ -137,39 +137,46 @@ def test_full_bad_filled_value(self):
     (np.arange(24).reshape(4, 3, 2), "f4"),
 ]
 LIKE_FUNCTIONS = ("zeros_like", "ones_like")
+SHAPE_ARG = (None, (-1,), (1, -1))
 
 
 @pytest.mark.parametrize("x_np,dtype", DATA_ARGS)
-def test_empty_like(x_np, dtype):
+@pytest.mark.parametrize("shape", SHAPE_ARG)
+def test_empty_like(x_np, dtype, shape):
+    shape = shape if shape is None else x_np.reshape(shape).shape
```

Comment:
I don't think you need to do this. `x_np.reshape(shape)` doesn't modify `x_np`, it creates a new view/copy. And `arr.reshape(s).shape` always returns `s`.

Reply:
It is just a convenience way to create compatible shapes. 
Notice, `arr.reshape(s).shape` makes NumPy/cuNumeric infer the value of the `-1` in `s`.

Reply:
Ah, ok, I missed the -1 in the shapes. You're right, those need to be sanitized.